### PR TITLE
Update docs ahead of new schema launch

### PR DESCRIPTION
**Pull request by samanvp**
This commit includes the following updates:
 * sharding.md
 * large_inputs.md

and adds a new page:
 * query_cost.md

### CODE REVIEW

**Comment by samanvp** [comment]
I am not sure if this is the best way to review documentation. If it makes things easier, we can work on a google doc and later I will copy the text back to this PR.

Please let me know.

File: docs/query_cost.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+for the genomics data where majority of the queries are analyzing a small genomic region.
+Table partitioning comes at no cost, and allows to divide a table up to 4000
+segments. Appending to a paritioned table will not degrade the performance and all newly
+added rows will be stored in the right partition. Also you can know query cost before
+running your query. Because partition pruning is done before the query runs, 
+you can get the query cost through a [dry run](https://cloud.google.com/bigquery/docs/dry-run-queries). 
+
+## (III) BigQuery Clustering
+
+[BigQuery clustering](https://cloud.google.com/bigquery/docs/clustered-tables)
```
Comment (kemp-google) [inline docs/query_cost.md:86]: I would drop all of this and just talk about the clustering we perform.  If they want to know more, they can read the BQ docs.

File: docs/query_cost.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+| ChrY       | 1,344,278       | 1.64       |
+| ChrX       | 28,156,663      | 40.51      |
+
+
+Other than making each table smaller and more manageable, sharding per chromosome
+is a prerequisite for the next cost reduction technique.
+
+## (II) BigQuery Partitioning
+ 
+BigQuery [partitioning](https://cloud.google.com/bigquery/docs/partitioned-tables#integer_range)
```
Comment (kemp-google) [inline docs/query_cost.md:58]: I would drop most of this, refer to the feature and just describe what we partition on.

File: docs/large_inputs.md [hunk 3]
```diff
@@ -99,51 +56,101 @@ transforms (e.g. the sample name is repeated in every record in the BigQuery
...
+-  output_table:
+     table_name_suffix: "chr1_1M_2M"
+     regions:
+       - "chr1:1,000,000-2,000,000"
+       - "1"
+     partition_range_end: 2,000,000
+```
+
+### Saving AVRO files
+If you are processing large inputs, you could set the `--keep_intermediate_avro_files`
```
[the remark this answers is not in the document]
Comment (samanvp) [reply docs/large_inputs.md:150]: I agree, I thought at least for the first couple of month we recommend this. As soon as we realized our code is robust we will remove this.

If you don't mind, I will remove this section when #582 is resolved.

File: docs/query_cost.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+| Chr1       | 55,215,812      | 88.31      |
+| Chr2       | 58,605,556      | 94.01      |
+| Chr3       | 47,688,920      | 77.78      |
+| Chr21      | 10,248,793      | 15.62      |
+| Chr22      | 10,882,958      | 17.78      |
+| ChrY       | 1,344,278       | 1.64       |
+| ChrX       | 28,156,663      | 40.51      |
+
+
+Other than making each table smaller and more manageable, sharding per chromosome
```
[the remark this answers is not in the document]
Comment (samanvp) [reply docs/query_cost.md]: done.

File: docs/query_cost.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+if we are querying a small region of genome. 
+
+Partitioning is the perfect solution here: BigQuery considers the `WHERE` clause
+and based on that it only processes the set of partitioning that contain rows that
+are relevant to the query. Following figure schematically shows the cost of previous
+query on a table which is partitioned based on of `start_position` column :
+
+![query cost unpartitioned table](images/query_cost_partitioned_table.svg)
+
+Partitioning is a very effective technique to reduce the cost of queries, specially
```
[the remark this answers is not in the document]
Comment (samanvp) [reply docs/query_cost.md]: done.

File: docs/query_cost.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+for the genomics data where majority of the queries are analyzing a small genomic region.
```
[the remark this answers is not in the document]
Comment (samanvp) [reply docs/query_cost.md]: done.

File: docs/query_cost.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+Table partitioning comes at no cost, and allows to divide a table up to 4000
```
[the remark this answers is not in the document]
Comment (samanvp) [reply docs/query_cost.md]: done.

File: docs/query_cost.md (hunk 2 above)
Comment (samanvp) [reply docs/query_cost.md:58]: There is no good partitioning doc available. I really like these two images to explain how partitioning work.

File: docs/query_cost.md (hunk 1 above)
Comment (samanvp) [reply docs/query_cost.md:86]: I agree, we better keep this section as small as possible and refer users to BigQuery docs.
If you don't mind I will leave this for Ben to do.

File: docs/large_inputs.md [hunk 8]
```diff
@@ -1,149 +1,172 @@
...
-millions of samples, and billions of records. There are a few settings that
-may need to be adjusted depending on the size of the input files. Each of these
-settings are explained in the sections below.
+The Variant Transforms pipeline can process hundreds of thousands of files,
+millions of samples, and billions of records. There are a few settings that 
+may need to be adjusted depending on the size of the input files. Each of 
+these settings are explained in the sections below.
 
-Default settings:
+Example usage:
```
Comment (kemp-google) [inline docs/large_inputs.md:8]: Can we drop the flags that have sensible defaults ?

And just show what flags they need to provide?

**Pull request merged**